inference/predictor_deployment.py

```diff
@@ -94,23 +100,37 @@ async def __call__(self, http_request: Request) -> Union[StreamingResponse, str]
                 prompts.extend(text)
         else:
             prompts.append(text)
+
         if not streaming_response:
-            return self.predictor.generate(prompts, **config)
+            if self.use_vllm:
+                return await self.predictor.generate_async(prompts, **config)
+            else:
+                return self.predictor.generate(prompts, **config)
+
         if self.use_deepspeed:
             self.predictor.streaming_generate(prompts, self.streamer, **config)
             return StreamingResponse(
                 self.consume_streamer(), status_code=200, media_type="text/plain"
             )
+        elif self.use_vllm:
+            # TODO: streaming only support single prompt
+            # It's a wordaround for current situation, need another PR to address this
```

Comment:
What causes the limitation and what needs to be done to address it?

Reply:
> What causes the limitation and what needs to be done to address it?

Another PR will change the protocol for #36, then we will error here since we didn't support streaming for single http request including multiple prompts.